tensorflow_model_optimization/g3doc/guide/clustering/index.md

```diff
@@ -0,0 +1,103 @@
+# Weight Clustering
+
+This document provides an overview on weight clustering to help you determine how it fits with your use case. To dive right into the code, see the [Clustering with Keras](clustering_with_keras.ipynb) tutorial and the [API docs](../../api_docs/python). For additional details on how to use the Keras API, a deep dive into weight clustering, and documentation on more advanced usage patterns, see the [API usage guide](train_clustered_models.md).
+
+## Overview
+
+Clustering, or weight sharing, is a technique for reducing the number of unique weight values in a model. It first groups the weights of each layer into *N* clusters, then shares the cluster's centroid value for all the weights belonging to the cluster.
```

Comment:
For conciseness: -> "Clustering, or weight sharing, reduces the number of unique weight values in a model."

"Clustering, or weight sharing, reduces the number of unique weight values in a model, leading to benefits for deployment." to mirror the quantization docs